dispatch/static/manager/src/js/components/fields/IntegerField.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+
+import { TextInput } from '../inputs'
+
+export default class IntegerField extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      selectionStart: 0,
+      selectionEnd: 0
+    }
+  }
+
+  onChange(val) {
+    let { selectionStart, selectionEnd } = this.refs.input.refs.input
+    const initialLength = val.length
+
+    val = val.replace(/[^\d-]+/g, '')
+
+    // allow a '-' at start, but nowhere else
```

Comment:
one option is something like this, which could be further reduced into one line if you really wanted.
```js
const prefix = val[0] === '-' ? '-' : ''
val = prefix + val.replace(/[^\d]/g, '')
```